File: memories/2023/exercise/php/day12/src/Greeter.php
<?php

declare(strict_types = 1);

namespace Greeting;

class Greeter
{
	private $formality;

	public function greet(): string
	{
		if (null === $this->formality) {
			return 'Hello.';
		}

		if ('formal' === $this->formality) {
			return 'Good evening, sir.';
		} elseif ('casual' === $this->formality) {
			return 'Sup bro?';
		} elseif ('intimate' === $this->formality) {
			return 'Hello Darling!';
		} else {
			return 'Hello.';
		}
	}

	public function setFormality(string $formality): void
	{
		$this->formality = $formality;
	}
}

File: memories/2023/exercise/scala/day12/src/main/scala/greeting/Greeter.scala
package greeting

class Greeter {
  private var formality: String = _

  def greet(): String = formality match {
    case null => "Hello."
    case "formal" => "Good evening, sir."
    case "casual" => "Sup bro?"
    case "intimate" => "Hello Darling!"
    case _ => "Hello."
  }

  def setFormality(formality: String): Unit = {
    this.formality = formality
  }
}
